extensions/eclipse-che-theia-plugin-ext/src/browser/plugin/che-plugin-frontend-service.ts

```diff
@@ -0,0 +1,146 @@
+/********************************************************************************
+ * Copyright (C) 2019 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable, inject } from 'inversify';
+import {
+    HostedPluginServer,
+    PluginMetadata
+} from '@theia/plugin-ext/lib/common/plugin-protocol';
+
+import {
+    ChePluginMetadata
+} from '../../common/che-protocol';
+
+@injectable()
+export class ChePluginFrontentService {
+
+    @inject(HostedPluginServer)
+    protected readonly hostedPluginServer: HostedPluginServer;
+
+    async getDeployedPlugins(filter: string): Promise<ChePluginMetadata[]> {
+        if (this.hasType(filter, '@installed')) {
+            let pluginList = await this.getAllDeployedPlugins();
+            pluginList = this.filter(pluginList, filter);
+            return pluginList;
+        }
+
+        return [];
+    }
+
+    // @installed
+    // @builtin
+    // @enabled
+    // @disabled
+    hasType(filter: string, type: string) {
+        if (filter) {
+            const filters = filter.split(' ');
+            const found = filters.find(value => value === type);
+            return found !== undefined;
+        }
+
+        return false;
+    }
+
+    filterByType(plugins: ChePluginMetadata[], type: string): ChePluginMetadata[] {
+        return plugins.filter(plugin => {
+            const regex = / /gi;
+            const t = plugin.type.toLowerCase().replace(regex, '_');
+            return t === type;
+        });
+    }
+
+    filterByText(plugins: ChePluginMetadata[], text: string): ChePluginMetadata[] {
+        return plugins;
+    }
+
+    filter(plugins: ChePluginMetadata[], filter: string): ChePluginMetadata[] {
```

Comment:
Maybe move this method into common section? ([duplicate method](https://github.com/eclipse/che-theia/pull/162/files#diff-73b9783016d6d093aad90633e33edfdeR122))

Reply:
Yes. I have extracted it to the `common/plugin/plugin-filter`. Thanks!